javascript/ql/src/semmle/javascript/frameworks/Babel.qll

```diff
@@ -188,4 +188,20 @@ module Babel {
     /** Gets the name of the variable used to create JSX elements. */
     string getJsxFactoryVariableName() { result = getOption("pragma").(JSONString).getValue() }
   }
+
+  /**
+   * A taint step through a call to the Babel `transform` function.
+   */
+  private class TransformTaintStep extends TaintTracking::SharedTaintStep {
+    override predicate step(DataFlow::Node pred, DataFlow::Node succ) {
+      exists(DataFlow::CallNode call |
+        call =
+          API::moduleImport(["@babel/standalone", "@babel/core"])
+              .getMember(["transform", "transformSync"])
```

Comment:
is `.transform` async? Won't that make the `succ` turn up as a parameter to a callback?

Reply:
Ah you're right. I thought `transform` would return a promise but after a closer look it just becomes synchronous if you don't give it a callback. There's also `transformAsync` which _does_ return a promise.

Since neither `transformSync` nor `transformAsync` accept a third parameter I just added an extra case for the callback parameter.